### PR TITLE
Ticket CRUD

### PR DESCRIPTION
**Pull request by kwm4385**
fixes https://github.com/PagerNation/PagerNation/issues/58
CRUD Operations for tickets

### CODE REVIEW

File: server/tests/services/ticket.test.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  };
+
+  // Create
+  describe('createTicket()', () => {
+    context('with valid ticket details', () => {
+      it('creates a new ticket', (done) => {
+        ticketService.createTicket(ticketObject)
+          .then(createdTicket => {
+            expect(createdTicket).to.exist;
+            expect(createdTicket.groupId.toString()).to.equal('551137c2f9e1fac808a5f572');
```
Comment (paradime) [inline server/tests/services/ticket.test.js]: we should refer to ticket_object.groupId here

File: server/services/ticket.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        resolve(Ticket.findById(value));
+      }
+    });
+  });
+}
+
+function createTicket(ticketDetails) {
+  return new Promise((resolve, reject) => {
+    Joi.validate(ticketDetails, ticketSchema, (err, value) => {
+      if (err) {
```
Comment (paradime) [inline server/services/ticket.js]: We can replace these conditional blocks with
```javascript
if(err) return reject(err);
resolve(Ticket.create(value);
```

**Comment by paradime** [comment]
Noice!

File: server/services/ticket.js (hunk 2 above)
Comment (kwm4385) [inline server/services/ticket.js]: Done

Comment (jarrydlee) [inline server/services/ticket.js]: I'm not sure if I entirely agree with removing parens here. Is this a javascript convention that is widely used? If so could you link it @paradime?

**Pull request merged**